valkyrie/lib/valkyrie/storage_adapter.rb

```diff
@@ -44,6 +44,23 @@ class File < Dry::Struct
       def stream
         io
       end
+
+      def disk_path
```

Comment:
I wonder if we should start returning a different class from the Disk adapter vs, Fedora, so we don't need this conditional here.